app/feature/feature-movingflow/src/main/kotlin/com/hedvig/android/feature/movingflow/ui/chosecoveragelevelanddeductible/ChoseCoverageLevelAndDeductibleViewModel.kt

```diff
@@ -0,0 +1,194 @@
+import com.hedvig.android.feature.movingflow.ui.chosecoveragelevelanddeductible.DeductibleOptions.OneOption
+import com.hedvig.android.molecule.android.MoleculeViewModel
+import com.hedvig.android.molecule.public.MoleculePresenter
+import com.hedvig.android.molecule.public.MoleculePresenterScope
+import kotlinx.coroutines.flow.collectLatest
+
+internal class ChoseCoverageLevelAndDeductibleViewModel(
+  movingFlowRepository: MovingFlowRepository,
+) : MoleculeViewModel<ChoseCoverageLevelAndDeductibleEvent, ChoseCoverageLevelAndDeductibleUiState>(
+    ChoseCoverageLevelAndDeductibleUiState.Loading,
+    ChoseCoverageLevelAndDeductiblePresenter(movingFlowRepository),
+  )
+
+private class ChoseCoverageLevelAndDeductiblePresenter(
+  private val movingFlowRepository: MovingFlowRepository,
+) : MoleculePresenter<ChoseCoverageLevelAndDeductibleEvent, ChoseCoverageLevelAndDeductibleUiState> {
+  @Composable
+  override fun MoleculePresenterScope<ChoseCoverageLevelAndDeductibleEvent>.present(
+    lastState: ChoseCoverageLevelAndDeductibleUiState,
+  ): ChoseCoverageLevelAndDeductibleUiState {
+    var tiersInfo: Option<TiersInfo?> by remember {
+      mutableStateOf(
+        (lastState as? ChoseCoverageLevelAndDeductibleUiState.Content)?.tiersInfo?.some() ?: None,
+      )
+    }
+
+    CollectEvents { event ->
+      when (event) {
+        is SelectCoverage -> {
+          val currentContent = tiersInfo.getOrNull() ?: return@CollectEvents
+          val newSelectedCoverage =
+            currentContent.allOptions.firstOrNull { it.id == event.homeQuoteId } ?: return@CollectEvents
+          if (newSelectedCoverage == currentContent.selectedCoverage) return@CollectEvents
+          tiersInfo = currentContent.copy(
+            selectedCoverage = newSelectedCoverage,
+            selectedDeductible = null,
+          ).some()
+        }
+
+        is SelectDeductible -> {
+          val currentContent = tiersInfo.getOrNull() ?: return@CollectEvents
+          val newSelectedDeductible =
+            currentContent.allOptions.firstOrNull { it.id == event.homeQuoteId } ?: return@CollectEvents
+          tiersInfo = currentContent.copy(selectedDeductible = newSelectedDeductible).some()
+        }
+      }
+    }
+
+    LaunchedEffect(Unit) {
+      movingFlowRepository.movingFlowState().collectLatest { movingFlowState ->
+        if (movingFlowState?.movingFlowQuotes == null) {
+          tiersInfo = null.some()
+          return@collectLatest
+        }
+        val homeQuotes = movingFlowState.movingFlowQuotes.homeQuotes
+        if (homeQuotes.isEmpty()) {
+          tiersInfo = null.some()
+          return@collectLatest
+        }
+        val initiallySelectedHomeQuote = homeQuotes.firstNotNullOfOrNull { moveHomeQuote ->
+          if (moveHomeQuote.defaultChoice) moveHomeQuote else null
+        }
+        val uniqueCoverageOptions = homeQuotes
+          .groupBy { it.tierLevel }
```

Comment:
I think it should be tierName here

Reply:
Yup, it was part of the notes I kept when we were testing yesterday, I've fixed it in a separate commit now